fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/BindExpression.java

```diff
@@ -624,8 +631,34 @@ protected boolean condition(Rule rule, Plan plan) {
                         Slot boundSlot = new SlotReference(slot.getNameParts().get(1), generator.getDataType(),
                                 generator.nullable(), ImmutableList.of(slot.getNameParts().get(0)));
                         slotBuilder.add(boundSlot);
+                        // the boundSlot may has two situation:
+                        // 1. the expandColumnsAlias is not empty, we should use make boundSlot expand to multi alias
+                        // 2. the expandColumnsAlias is empty, we should use origin boundSlot
+                        if (generate.getExpandColumnAlias() != null
+                                && !CollectionUtils.isEmpty(generate.getExpandColumnAlias().get(i))) {
```

Comment:
`generate.getExpandColumnAlias().get(i)` could generate OutOfBound exception, should check first